tests/compendia_tests.py

```diff
@@ -0,0 +1,135 @@
+
+compendia_object_1 = {
+    "id": 69,
+    "primary_organism_name": "HUMAN",
+    "organism_names": [
+        "HUMAN",
+        "GORILLA"
+    ],
+    "svd_algorithm": "NONE",
+    "quant_sf_only": True,
+    "compendium_version": 1,
+    "computed_file": {
+        "id": 420,
+        "filename": "test-compendia.zip",
+        "size_in_bytes": 5000,
+        "is_smashable": False,
+        "is_qc": False,
+        "sha1": "3c5de2dba42c2768e7542a0262d2b9a1e9fc9f45",
+        "s3_bucket": "test-bucket",
+        "s3_key": "test-compendia.zip",
+        "created_at": "2020-10-02:00:00.0Z",
+        "last_modified": "2020-10-02:00:00.0Z"
+    }
+}
+
+compendia_object_2 = {
+    "id": 42,
+    "primary_organism_name": "GORILLA",
+    "organism_names": [
+        "HUMAN",
+        "GORILLA"
+    ],
+    "svd_algorithm": "NONE",
+    "quant_sf_only": True,
+    "compendium_version": 1,
+    "computed_file": {
+        "id": 4,
+        "filename": "test-compendia-2.zip",
+        "size_in_bytes": 7500,
+        "is_smashable": True,
+        "is_qc": False,
+        "sha1": "3c5de2dba42c2768e7542a0262d2b9a1e9fc9f45",
+        "s3_bucket": "test-bucket",
+        "s3_key": "test-compendia-2.zip",
+        "created_at": "2020-10-05:00:00.0Z",
+        "last_modified": "2020-10-05:00:00.0Z"
+    }
+}
+
+search_1 = {
+    "count": 2,
+    "next": "search_2",
+    "previous": None,
+    "results": [compendia_object_1]
+}
+
+search_2 = {
+    "count": 2,
+    "next": None,
+    "previous": "search_1",
+    "results": [compendia_object_2]
+}
+
+def mock_get_request(url, **kwargs):
+
+    class MockResponse:
+        def __init__(self, json_data):
+            self.json_data = json_data
+
+        def json(self):
+            return self.json_data
+        
+        def raise_for_status(self):
+            pass
+
+    if url == "https://api.refine.bio/v1/compendia/1/":
+        return MockResponse(compendia_object_1)
+
+    if url == "https://api.refine.bio/v1/compendia/":
+        return MockResponse(search_1)
+
+    if url == "search_2":
+        return MockResponse(search_2)
+
+class CompendiaTests(unittest.TestCase):
+
+    @patch("pyrefinebio.http.requests.get", side_effect=mock_get_request)
+    def test_compendia_get(self, mock_get):
+        result = pyrefinebio.Compendia.get(1)
+        self.assertCompendia(result, compendia_object_1)
+
+    @patch("pyrefinebio.http.requests.get", side_effect=mock_get_request)
+    def test_compendia_search_no_filters(self, mock_get):
+        result = pyrefinebio.Compendia.search()
+
+        result_list = list(result)
+
+        self.assertCompendia(result_list[0], compendia_object_1)
+        self.assertCompendia(result_list[1], compendia_object_2)
+
+        self.assertEqual(len(mock_get.call_args_list), 2)
+
+
+    def assertCompendia(self, actual, expected):
+        # Assert that feilds returned are properly set and can be accessed by dot notation
+        self.assertEqual(actual.id, expected["id"])
```

Comment:
I think I like this testing strategy! However this looks fairly manual to write out for every field of every object.

I think this might be a bit easier:
```python
def assertCompendia(self, actual: Compendia, expected: dict):
  for key, value in expected.items():
    self.assertEqual(getattr(actual, key), value)
```